marketing/sailthruScriptJob.groovy

```diff
@@ -0,0 +1,128 @@
+Map secretMap = [:]
+out.println('Parsing secret YAML file')
+try {
+    /* Parse k:v pairs from the secret file referenced by secretFileVariable */
+    String contents = new File("${EDX_SAILTHRU_SYNC_CONTENT}").text
+    Yaml yaml = new Yaml()
+    secretMap = yaml.load(contents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    assert jobConfig.containsKey('oauth_host')
+    assert jobConfig.containsKey('oauth_key')
+    assert jobConfig.containsKey('oauth_secret')
+    assert jobConfig.containsKey('sailthru_key')
+    assert jobConfig.containsKey('sailthru_secret')
+    assert jobConfig.containsKey('content_api_url')
+    assert jobConfig.containsKey('lms_url')
+    assert jobConfig.containsKey('report_email')
+    assert jobConfig.containsKey('type')
+    assert jobConfig.containsKey('fixups')
+    assert jobConfig.containsKey('mode')
+    assert jobConfig.containsKey('requirements')
+    assert jobConfig.containsKey('scriptPath')
+    assert jobConfig.containsKey('githubRepoUrlSegment')
+    assert jobConfig.containsKey('jobName')
+    assert jobConfig.containsKey('notificationEmail')
+
+    job(jobConfig['jobName']) {
+
+        requirements = "sudo pip install -r " + jobConfig['requirements']
+        sync_content = "python " + jobConfig['scriptPath'] + " --oauth_host "+ jobConfig['oauth_host'] + 
+                  " --oauth_key " + jobConfig['oauth_key'] + " --oauth_secret " + jobConfig['oauth_secret'] + 
+                  " --sailthru_key " + jobConfig['sailthru_key'] + " --sailthru_secret " + jobConfig['sailthru_secret'] +
+                  " --content_api_url " + jobConfig['content_api_url'] + " --lms_url " + jobConfig['lms_url'] +
+                  " --report_email " + jobConfig['report_email'] + " --fixups=" + jobConfig['fixups'] +
+                  " --type " + jobConfig['type'] + " " + jobConfig['mode']
+
+        properties {
+              githubProjectUrl(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['githubRepoUrlSegment'])
+        }
+
+        logRotator JENKINS_PUBLIC_LOG_ROTATOR() //Discard build after 14 days
+
+        label('master') //restrict to jenkins-worker
+
+        checkoutRetryCount(5)
+
+        scm {
+
+           git { //using git on the branch and url, clone, clean before checkout
+                remote {
+                    url(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['githubRepoUrlSegment'] + '.git')
+                }
+                branch('*/master')
+                browser()
+            }
+        }
+
+        steps {
+            shell( "sudo rm -rf /usr/local/lib/python2.7/dist-packages/requests*")
```

Comment:
Jenkins does not have sudo access.
This code isn't using a virtualenv, which is why it's failling.

please look at something like Renzo's recent work https://github.com/edx-ops/edx-jenkins-job-dsl/blob/master/jobs/tools-edx-jenkins.edx.org/createTranslationJobs.groovy to get a sense of how this sort of thing should be configured.  Think about whether this is an internal job or something that should be public.  Think about credential management.  Also, we don't use scm, we use multiscm.  That's just the most basic review from what I saw in here around sudo, I don't have cycles to look at the rest of this PR right now.